### ISSUE TITLE
bpftrace -l should list software and hardware types

### ISSUE THREAD

**Issue by brendangregg** [iovisor/bpftrace#44]
The software and hardware probe types are not listed in -l. They should be.

This might also be a chance to refactor code like this, from attached_probe.cpp:

```C++
  else if (probe_.path == "cache-misses")
  {
    type = PERF_COUNT_HW_CACHE_MISSES;
  }
  else if (probe_.path == "branch-instructions" || probe_.path == "branches")
  {
    type = PERF_COUNT_HW_BRANCH_INSTRUCTIONS;
    defaultp = 100000;
  }
  else if (probe_.path == "bus-cycles")
  {
    type = PERF_COUNT_HW_BUS_CYCLES;
    defaultp = 100000;
  }
```

Perhaps there should just be one lookup table that had the types, paths, and counts, and all other code (including the new listing code) can refer to that one lookup table.

**Comment by brendangregg** [comment]
fixed by #352, closing.

**Comment by brendangregg** [comment]
I'm switching closed 1.0 issues to 0.9, so that we have a neat summary of what's in the 0.9 release:

https://github.com/iovisor/bpftrace/milestone/1?closed=1
